### PR TITLE
Date picker for Timeline UI

### LINKED ISSUE #4320: Date picker for Timeline UI
### 💻 Environment

Platform: Windows

### 📒 Description
Implement date picker for Timeline UI. Part of #4269

### PR DESCRIPTION
**Pull request by atsvirchkova**
### 📒 Description
Date picker for Timeline UI

### 🕶️ Types of changes

 - **New feature** (non-breaking change which adds functionality)

### 🤯 List of changes
src/ui/windows/TogglDesktop/TogglDesktop/Toggl.cs
src/ui/windows/TogglDesktop/TogglDesktop/ui/Resources/DesignUpdate/DatePicker.xaml
src/ui/windows/TogglDesktop/TogglDesktop/ui/Resources/DesignUpdate/Icons.xaml
src/ui/windows/TogglDesktop/TogglDesktop/ui/ViewModels/TimelineViewModel.cs
src/ui/windows/TogglDesktop/TogglDesktop/ui/views/EditView.xaml
src/ui/windows/TogglDesktop/TogglDesktop/ui/views/Timeline.xaml

### 👫 Relationships

Closes #4320 

### 🔎 Review hints
I've spent some time trying to fit the design for date picker. There are still details that might need polishing:
- The color when hovering over date picker buttons is too dark. I'm actually not sure, that we need any color there.
- I turned off the ability to enter date manually with keyboard for now.
- There is somewhat not logical behavior with calendar popup, when user clicks on the button while the popup is opened, it doesn't close the popup. The popup is only closed when user clicks outside the button.

### CODE REVIEW

File: src/ui/windows/TogglDesktop/TogglDesktop/ui/Resources/DesignUpdate/Icons.xaml [hunk 1]
```diff
@@ -78,4 +78,54 @@
         <Rectangle Width="14" Height="14" RadiusX="7" RadiusY="7" Fill="{StaticResource Toggl.AccentBrush}" StrokeThickness="1"/>
         <Path Canvas.Top="3.2" Canvas.Left="2.5" Fill="{DynamicResource Toggl.CardBackground}" StrokeThickness="1" Data="M3.75914017 5.02308449 L7.00866193 0.411902318 C7.30820656 -0.0131617185 7.91527881 -0.128022547 8.36459575 0.155353478 C8.81391269 0.438729502 8.93532714 1.01303364 8.63578251 1.43809768 L4.7246714 6.98809768 C4.38240853 7.47378037 3.65601908 7.54182469 3.21971781 7.12907377 L0.77527337 4.81657377 C0.393427395 4.4553388 0.393427395 3.8696612 0.77527337 3.50842623 C1.15711934 3.14719126 1.77621399 3.14719126 2.15805996 3.50842623 L3.75914017 5.02308449 Z"/>
     </Canvas>
+    <Viewbox Stretch="Uniform" x:Key="Toggl.DatePickerLeft">
```
Comment (skel35) [inline src/ui/windows/TogglDesktop/TogglDesktop/ui/Resources/DesignUpdate/Icons.xaml:81]: I guess you just copy-pasted it from an icon converter tool so it is OK considering it's just an icon, but I find it nice that the whole thing can be simplified to:
```
    <Viewbox Stretch="Uniform" x:Key="Toggl.DatePickerLeft">
        <Path Width="24" Height="25" Fill="#FF333333" Data="M9.902 13l5.362-5.361-.528-.528L8.848 13l5.888 5.889.528-.528z" />
    </Viewbox>
```

**Comment by atsvirchkova** [comment]
Fixed the dark color mode and created attached properties for Previous and Next buttons in date picker. Also fixed the calendar popup behavior.

**Comment by skel35** [comment]
I've just rebuilt and ran the app, clicked the Left button on the DatePicker and the app crashed:
>System.ArgumentOutOfRangeException: Value to add was out of range.
Parameter name: value
   at System.DateTime.Add(Double value, Int32 scale)
   at System.DateTime.AddSeconds(Double value)
   at System.DateTimeOffset.AddSeconds(Double seconds)
   at TogglDesktop.Toggl.DateTimeFromUnix(UInt64 unix_seconds) in C:\dev\toggl\toggldesktop\src\ui\windows\TogglDesktop\TogglDesktop\Toggl.cs:line 1439
   at TogglDesktop.ViewModels.TimelineViewModel.HandleDisplayTimeline(Boolean open, String date, List`1 first, List`1 firstTimeEntry, Int64 startDay, Int64 endDay) in C:\dev\toggl\toggldesktop\src\ui\windows\TogglDesktop\TogglDesktop\ui\ViewModels\TimelineViewModel.cs:line 29
   at TogglDesktop.Toggl.DisplayTimeline.Invoke(Boolean open, String date, List`1 first, List`1 firstTimeEntry, Int64 startDay, Int64 endDay)
   at TogglDesktop.Toggl.<>c.<listenToLibEvents>b__492_30(Boolean open, String date, IntPtr first, IntPtr firstTimeEntry, Int64 startDay, Int64 endDay) in C:\dev\toggl\toggldesk

**Comment by atsvirchkova** [comment]
Changed the types in DisplayTimeline delgate. Please, check if it works now. :pray:

**Review by skel35 — APPROVED** [review]
Works fine now 👍

**Pull request merged**